### ISSUE TITLE
自定义字典，loss: 1626.304443，acc：0.000000

### ISSUE THREAD

**Issue by dengmingD** [PaddlePaddle/PaddleOCR#5646]
[2022/03/07 15:38:03] root INFO: epoch: [1/500], iter: 100, lr: 0.000000, loss_ctc_Student_0: 782.231079, loss_ctc_Teacher_1: 779.671387, dml_0: 65.562668, loss: 1627.260986, loss_distance_l2_Student_Teacher_0: 0.020547, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, re
ader_cost: 0.02165 s, batch_cost: 0.28587 s, samples: 6464, ips: 226.12003
[2022/03/07 15:39:29] root INFO: epoch: [1/500], iter: 200, lr: 0.000000, loss_ctc_Student_0: 782.026733, loss_ctc_Teacher_1: 779.686768, dml_0: 65.675194, loss: 1626.855469, loss_distance_l2_Student_Teacher_0: 0.020589, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, re
ader_cost: 0.05396 s, batch_cost: 0.27715 s, samples: 6400, ips: 230.92427
[2022/03/07 15:41:04] root INFO: epoch: [1/500], iter: 300, lr: 0.000000, loss_ctc_Student_0: 780.862671, loss_ctc_Teacher_1: 778.191040, dml_0: 65.982422, loss: 1625.391602, loss_distance_l2_Student_Teacher_0: 0.020808, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, re
ader_cost: 0.14439 s, batch_cost: 0.35698 s, samples: 6400, ips: 179.28134
[2022/03/07 15:42:41] root INFO: epoch: [1/500], iter: 400, lr: 0.000000, loss_ctc_Student_0: 782.306580, loss_ctc_Teacher_1: 778.961182, dml_0: 66.114265, loss: 1626.304443, loss_distance_l2_Student_Teacher_0: 0.020567, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, re
ader_cost: 0.14260 s, batch_cost: 0.35796 s, samples: 6400, ips: 178.78953

**Comment by andyjpaddle** [comment]
方便提供一下用的配置文件和所做的修改细节吗？

**Comment by dengmingD** [comment]
Global:
  debug: false
  use_gpu: true
  epoch_num: 500
  log_smooth_window: 100
  print_batch_step: 100
  save_model_dir: ./output/rec_pp-OCRv2_distillation_1_5_2k
  save_epoch_step: 1
  eval_batch_step: [0, 200000]
  cal_metric_during_train: true
  pretrained_model: ./train_model/ch_PP-OCRv2_rec_train/best_accuracy
  checkpoints:
  save_inference_dir:
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ppocr/utils/ppocr_keys_v2.txt
  character_type: ch
  max_text_length: 25
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_pp-OCRv2_distillation.txt


Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Piecewise
    decay_epochs : [700, 800]
    values : [0.0001, 0.00001]
    warmup_epoch: 5
  regularizer:
    name: L2
    factor: 2.0e-06

Architecture:
  model_type: &model_type "rec"
  name: DistillationModel
  algorithm: Distillation
  Models:
    Teacher:
      pretrained:
      freeze_params: false
      return_all_feats: true
      model_type: *model_type
      algorithm: CRNN
      Transform:
      Backbone:
        name: MobileNetV1Enhance
        scale: 0.5
      Neck:
        name: SequenceEncoder
        encoder_type: rnn
        hidden_size: 64
      Head:
        name: CTCHead
        mid_channels: 96
        fc_decay: 0.00002
    Student:
      pretrained:
      freeze_params: false
      return_all_feats: true
      model_type: *model_type
      algorithm: CRNN
      Transform:
      Backbone:
        name: MobileNetV1Enhance
        scale: 0.5
      Neck:
        name: SequenceEncoder
        encoder_type: rnn
        hidden_size: 64
      Head:
        name: CTCHead
        mid_channels: 96
        fc_decay: 0.00002


Loss:
  name: CombinedLoss
  loss_config_list:
  - DistillationCTCLoss:
      weight: 1.0
      model_name_list: ["Student", "Teacher"]
      key: head_out
  - DistillationDMLLoss:
      weight: 1.0
      act: "softmax"
      use_log: true
      model_name_pairs:
      - ["Student", "Teacher"]
      key: head_out
  - DistillationDistanceLoss:
      weight: 1.0
      mode: "l2"
      model_name_pairs:
      - ["Student", "Teacher"]
      key: backbone_out

PostProcess:
  name: DistillationCTCLabelDecode
  model_name: ["Student", "Teacher"]
  key: head_out

Metric:
  name: DistillationMetric
  base_metric_name: RecMetric
  main_indicator: acc
  key: "Student"

Train:
  dataset:
    name: SimpleDataSet
    data_dir: d:/ocr/train_data_1_5_2k
    label_file_list:
    - d:/ocr/train_data_1_5_2k/lable.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecAug:
    - CTCLabelEncode:
    - RecResizeImg:
        image_shape: [3, 32, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: true
    batch_size_per_card: 64
    drop_last: true
    num_sections: 1
    num_workers: 1
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: d:/ocr/test_data
    label_file_list:
    - d:/ocr/test_lable.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - CTCLabelEncode:
    - RecResizeImg:
        image_shape: [3, 32, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 64
    num_workers: 1

**Comment by andyjpaddle** [comment]
这边是只更换了字典吗

**Comment by dengmingD** [comment]
是

**Comment by andyjpaddle** [comment]
这边训练多少了？我看截图才第一个epoch，往后跑跑再看看

**Comment by Babysun-ljh** [comment]
修改了字典相当于重新训练模型，刚开始acc=0是正常的

**Comment by dengmingD** [comment]
[2022/03/08 08:17:05] root INFO: epoch: [1/500], iter: 60800, lr: 0.000524, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.20
600 s, batch_cost: 0.41460 s, samples: 6400, ips: 154.36694
[2022/03/08 08:18:40] root INFO: epoch: [1/500], iter: 60900, lr: 0.000525, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.24
236 s, batch_cost: 0.46287 s, samples: 6400, ips: 138.26638
[2022/03/08 08:20:19] root INFO: epoch: [1/500], iter: 61000, lr: 0.000525, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.26
753 s, batch_cost: 0.49522 s, samples: 6400, ips: 129.23520
[2022/03/08 08:21:58] root INFO: epoch: [1/500], iter: 61100, lr: 0.000526, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.26
763 s, batch_cost: 0.49251 s, samples: 6400, ips: 129.94672
[2022/03/08 08:23:26] root INFO: epoch: [1/500], iter: 61200, lr: 0.000527, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.20
522 s, batch_cost: 0.40227 s, samples: 6400, ips: 159.09780

但loss: nnaxxx，数据1.48千万

**Comment by yjhtc007** [comment]
> [2022/03/08 08:17:05] root INFO: epoch: [1/500], iter: 60800, lr: 0.000524, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.20 600 s, batch_cost: 0.41460 s, samples: 6400, ips: 154.36694 [2022/03/08 08:18:40] root INFO: epoch: [1/500], iter: 60900, lr: 0.000525, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.24 236 s, batch_cost: 0.46287 s, samples: 6400, ips: 138.26638 [2022/03/08 08:20:19] root INFO: epoch: [1/500], iter: 61000, lr: 0.000525, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.26 753 s, batch_cost: 0.49522 s, samples: 6400, ips: 129.23520 [2022/03/08 08:21:58] root INFO: epoch: [1/500], iter: 61100, lr: 0.000526, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.26 763 s, batch_cost: 0.49251 s, samples: 6400, ips: 129.94672 [2022/03/08 08:23:26] root INFO: epoch: [1/500], iter: 61200, lr: 0.000527, loss_ctc_Student_0: nanxxx, loss_ctc_Teacher_1: nanxxx, dml_0: nanxxx, loss: nanxxx, loss_distance_l2_Student_Teacher_0: nanxxx, acc: 0.000000, norm_edit_dis: 0.000016, Teacher_acc: 0.000000, Teacher_norm_edit_dis: 0.000016, reader_cost: 0.20 522 s, batch_cost: 0.40227 s, samples: 6400, ips: 159.09780
> 
> 但loss: nnaxxx，数据1.48千万

调一下学习率试试？

**Comment by dengmingD** [comment]
找到问题了，这个先关下，新提一个issue

**Comment by jolt2017** [comment]
> 找到问题了，这个先关下，新提一个issue:#5655

啥问题？我这边遇到一个先收敛，再发散的问题，搞不定
